### PR TITLE
Move codec-server and grpc-proxy to own module

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
Move codec-server and grpc-proxy to own module. This will let us decouple most of our samples dependencies from the sever dependencies.

### CODE REVIEW

**Comment by Quinn-With-Two-Ns** [comment]
closes https://github.com/temporalio/samples-go/issues/237

File: codec-server/codec-server/codec-server
Comment (cretz) [inline codec-server/codec-server/codec-server:1]: Accidental committed binary?

File: codec-server/go.mod [hunk 1]
```diff
@@ -0,0 +1,87 @@
+module github.com/temporalio/samples-go/codec-server
+
+go 1.21.1
+
+require (
+	github.com/golang/snappy v0.0.4
+	github.com/google/uuid v1.5.0
+	go.temporal.io/api v1.26.0
+	go.temporal.io/sdk v1.25.1
+	go.temporal.io/server v1.22.4
```
Comment (cretz) [inline codec-server/go.mod:10]: I don't think the codec server sample should have to depend on the server library at all (other SDKs get along fine without it, though they don't have OIDC, but surely we can copy stuff). But I understand if now is not the time you want to remove this dependency.

File: codec-server/codec-server/codec-server
Comment (Quinn-With-Two-Ns) [reply codec-server/codec-server/codec-server:1]: ah good catch, removed.

File: codec-server/go.mod (hunk 1 above)
Comment (Quinn-With-Two-Ns) [reply codec-server/go.mod:10]: Yes I would rather do any changes like that in a separate PR

**Review by cretz — APPROVED** [review]

**Pull request merged**